pandas/_libs/groupby.pyx

```diff
@@ -779,7 +779,7 @@ def group_quantile(ndarray[float64_t] out,
                 non_na_counts[lab] += 1
 
     # Get an index of values sorted by labels and then values
-    order = (values, labels)
+    order = (values, np.where(labels == -1, labels.max() + 1, labels))
```

Comment:
I think we want the missing labels lexsorted to the _back_ and not the front of the array